src/gsy_e/models/strategy/energy_parameters/heat_pump_tank.py

```diff
@@ -0,0 +1,311 @@
+        self._parameters = tank_parameters
+        self._state = HeatPumpTankState(
+            initial_temp_C=tank_parameters.initial_temp_C,
+            slot_length=slot_length,
+            min_storage_temp_C=tank_parameters.min_temp_C,
+            max_storage_temp_C=tank_parameters.max_temp_C,
+        )
+
+    @property
+    def _Q_specific(self):
+        return SPECIFIC_HEAT_CONST_WATER * self._parameters.tank_volume_L * WATER_DENSITY
+
+    def serialize(self):
+        """Serializable dict with the parameters of the water tank."""
+        return {
+            "max_temp_C": self._parameters.max_temp_C,
+            "min_temp_C": self._parameters.min_temp_C,
+            "tank_volume_l": self._parameters.tank_volume_L,
+        }
+
+    def increase_tank_temp_from_heat_energy(self, heat_energy: float, time_slot: DateTime):
+        """Increase the temperature of the water tank with the provided heat energy."""
+        temp_increase_K = self._Q_kWh_to_temp_diff(heat_energy)
+        self._state.update_temp_increase_K(time_slot, temp_increase_K)
+
+    def decrease_tank_temp_from_heat_energy(self, heat_energy: float, time_slot: DateTime):
+        """Decrease the temperature of the water tank with the provided heat energy."""
+        temp_decrease_K = self._Q_kWh_to_temp_diff(heat_energy)
+        self._state.set_temp_decrease_K(time_slot, temp_decrease_K)
+
+    def increase_tank_temp_from_temp_delta(self, temp_diff: float, time_slot: DateTime):
+        """Increase the tank temperature from temperature delta."""
+        self._state.update_temp_increase_K(time_slot, temp_diff)
+
+    def decrease_tank_temp_vhp(self, heat_energy: float, time_slot: DateTime):
+        """
+        Decrease the tank temperature. Return True if the operation incurs in unmatched demand.
+        """
+        temp_differential_per_sec = -heat_energy / (
+            WATER_DENSITY * WATER_SPECIFIC_HEAT_CAPACITY * self._parameters.tank_volume_L
+        )
+        temp_decrease_C = temp_differential_per_sec * GlobalConfig.slot_length.total_seconds()
+        # Temp decrease is a negative value, therefore we need to add it to the current temp.
+        new_temperature_without_operation_C = (
+            self._state.get_storage_temp_C(time_slot) + temp_decrease_C
+        )
+        if new_temperature_without_operation_C < self._parameters.min_temp_C:
+            # Tank temp drops below minimum. Setting zero to tank temperature and reporting the
+            # unmatched heat demand event in order to be calculated and tracked.
+            self._state.set_temp_decrease_K(time_slot, 0.0)
+            return True
+        assert temp_decrease_C <= 0.0
+        self._state.set_temp_decrease_K(time_slot, abs(temp_decrease_C))
+        return False
+
+    def get_max_energy_consumption(self, cop: float, time_slot: DateTime):
+        """Calculate max energy consumption that a heatpump with provided COP can consume."""
+        max_desired_temp_diff = (
```

Comment:
Here I find "desired" misleading. AFAIU, this method returns the maximal energy consumption regarding the temperature limits. If it is desired to read the max temp is not clear at this point right?
How about: `max_temp_diff` or `max_possible_temp_diff`?

Reply:
Agreed, this is a remnant from refactoring. Fixing